File: Server_trunk/res/scripts/cell/Spell/ObjectFinder/RoundFinder.py
# -*- coding: utf-8 -*-
#
"""
"""
from KBEDebug import *

from .ObjectFinder import ObjectFinder

class RoundFinder(ObjectFinder):
	"""
	圆形目标搜索器
	"""
	def find(self, finder, position = None):
		"""
		"""
		return finder.entitiesInRange( self.radius, None, position)

	def init(self, dataSection):
		"""
		virtual method.
		从配置中初始化
		"""
		# 搜索半径
		self.radius = dataSection.readFloat( "radius" )

File: Client_trunk2/Assets/Scripts/Spells/ObjectFinder/RoundFinder.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

namespace SPELL
{
    public class RoundFinder : FinderBase
    {
        float radius = 0.0f;

        public override void Init(DataSection.DataSection dataSection)
        {
            radius = dataSection.readFloat("radius");
        }

        public override List<AvatarComponent> Find(AvatarComponent src)
        {
            return AvatarComponent.AvatarInRange(radius, src, Vector3.zero);
        }
    }
}

